scripts/ci-boot-simulator.sh

```diff
@@ -14,37 +14,51 @@ set -euo pipefail
 # Parse named arguments
 XCODE_VERSION="16.2" # Default value
 
+# Only allow one argument type
+if [[ $# -gt 2 ]]; then
+    echo "Error: Cannot specify both --xcode and --device"
+    echo "Usage: $0 [-x|--xcode <version>] OR [-d|--device <device>]"
+    exit 1
+fi
+
```

Comment:
`m`: This needs to be remove the next time another param is added. Please instead do a check after the parsing of the the options to check it's XOR, and throw the error if both variables are set.